controller/admin_controller.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ *
+ * Ideas extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\controller;
+
+/**
+ * Admin controller
+ */
+class admin_controller implements admin_interface
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\language\language */
+	protected $language;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var array */
+	protected $cfg_array = array();
+
+	/** @var array */
+	protected $errors = array();
+
+	/** @var array */
+	protected $new_config = array();
+
+	/** @var string */
+	protected $phpbb_root_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var string */
+	public $u_action;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\config\config              $config               Config object
+	 * @param \phpbb\db\driver\driver_interface $db                   Database object
+	 * @param \phpbb\language\language          $language             Language object
+	 * @param \phpbb\log\log                    $log                  Log object
+	 * @param \phpbb\request\request            $request              Request object
+	 * @param \phpbb\template\template          $template             Template object
+	 * @param \phpbb\user                       $user                 User object
+	 * @param string                            $root_path            phpBB root path
+	 * @param string                            $php_ext              php_ext
+	 * @access public
+	 */
+	public function __construct(\phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\language\language $language, \phpbb\log\log $log, \phpbb\request\request $request, \phpbb\template\template $template, \phpbb\user $user, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function display_options($id, $mode)
+	{
+		// Create a form key for preventing CSRF attacks
+		add_form_key('acp_phpbb_ideas_settings');
+
+		$this->new_config = $this->config;
+		$this->cfg_array = ($this->request->is_set('config')) ? $this->request->variable('config', array('' => ''), true) : $this->new_config;
+
+		// Check if selected user exists
+		$this->check_ideas_topics_poster();
+
+		// Set Ideas forum  options and registered usergroup forum permissions
+		$this->set_ideas_forum_options($id, $mode);
```

Comment:
I would remove this. `set_ideas_forum_options()` could be called from the module with something like:
```php
if ($request->is_set_post('ideas_forum_setup'))
{
    $admin_controller->set_ideas_forum_options();
}
```
It probably also shouldn't need the mode or id vars.
Look at how board rules does this for the send notifications button:
https://github.com/phpbb-extensions/boardrules/blob/master/acp/boardrules_module.php#L47-L51
https://github.com/phpbb-extensions/boardrules/blob/master/controller/admin_controller.php#L572-L601

Also to get it working right, you can remove all the mode and id variables from the controller and the logging for this action should be done within the set_ideas_forum_options method, not inside the display options method. Basically, just follow the example of board rules linked to above. ;)